### PR TITLE
Allow SetpointManager:OutdoorAirReset to work with positive or negative slope

### PR DESCRIPTION
**Pull request by rglazer2**
Addresses #4627.
Previously, SetpointManager:OutdoorAirReset expected the "Setpoint at Outdoor Low Temperature" to be higher than the "Setpoint at Outdoor High Temperature". If that was not the case, the manager would silently use the average of the two values as a constant setpoint.  Now the manager works as expected either way.

### CODE REVIEW

**Comment by rglazer2** [comment]
@mjwitte Let me know how this looks.

File: src/EnergyPlus/SetPointManager.hh [hunk 1]
```diff
@@ -1266,6 +1266,10 @@ namespace SetPointManager {
 								 int const CompOpType,
 								 int const ControlNodeNum
 								 );
+
+	Real64
+	CalcSetPoint(Real64 OutLowTemp, Real64 OutHighTemp, Real64 OutDryBulbTemp, Real64 SetTempAtOutLow, Real64 SetTempAtOutHigh);
```
Comment (mjwitte) [inline src/EnergyPlus/SetPointManager.hh]: @rglazer2 This function is specific to the OA setpoint manager, so it belongs inside the struct DefineOutsideAirSetPointManager   See DefineIdealCondEntSetPointManager::setupSetPointAndFlags for example.  Also, preferred style is to stack arguments in the declaration.

**Comment by rglazer2** [comment]
@mjwitte Updated.

**Comment by mjwitte** [comment]
@rglazer2 That's good, thanks.  Now, please merge in the latest develop branch so we can get a clean test report.

**Comment by rglazer2** [comment]
@mjwitte All set.

**Comment by mjwitte** [comment]
@Myoldmopar This is ready for review.

**Comment by Myoldmopar** [comment]
Hmmm, so it's just really removing the check for the setpoint values?  The code looks good, thanks for the unit test.  I ran the example file in some variations with develop and this branch.  Things are good.  On the Winter design day when the outdoor air is -17.3C, here are the results:

| Branch | SP@low | SP@high | SP |
| --- | --- | --- | --- |
| develop | 80 | 90 | **85** |
| " | 91 | 90 | 91 |
| " | 90 | 90 | 90 |
| rglazer2:101163... | 80 | 90 | **80** |
| " | 91 | 90 | 91 |
| " | 90 | 90 | 90 |

This is good.  On a side note, boy it was sure nice to see the console stdout scrolling along beautifully throughout the simulation on the develop branch with my accidental commit.  :+1: With this bug fix branch it all just came out at once when the simulation was over.  :-1: 

OK, I'm merging.

**Pull request merged**